### PR TITLE
fix(driver): extend modern bpf test framework to all drivers (part 3)

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind bug

/kind feature

**Any specific area of the project related to this PR?**

/area driver-kmod

/area driver-bpf

/area driver-modern-bpf

**Does this PR require a change in the driver versions?**

No

**What this PR does / why we need it**:

This PR belongs to the https://github.com/falcosecurity/libs/pull/783 series. It fixes some differences between the 3 drivers and the usual issue of `fd` on 32 bit. It tries also to simplify code paths where possible.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: driver/ppm_flag_helpers.h [hunk 1]
```diff
@@ -888,14 +888,39 @@ static __always_inline u8 sockopt_optname_to_scap(int level, int optname)
...
+ * we couldn't be able to detect the right flag value, for example: 
+ * `SO_RCVTIMEO_OLD` is defined so we compile only this branch, but
+ * actual value of `SO_RCVTIMEO` is `SO_RCVTIMEO_NEW`.
+ * https://github.com/torvalds/linux/commit/a9beb86ae6e55bd92f38453c8623de60b8e5a308
+ */
 #ifdef SO_RCVTIMEO
 		case SO_RCVTIMEO:
 			return PPM_SOCKOPT_SO_RCVTIMEO;
 #endif
+#if (defined(SO_RCVTIMEO_OLD) && !defined(SO_RCVTIMEO)) || (defined(SO_RCVTIMEO_OLD) && (SO_RCVTIMEO_OLD != SO_RCVTIMEO))
```
Comment (FedeDP) [inline driver/ppm_flag_helpers.h:903]: This is a fix actually, right?

**Review by FedeDP — APPROVED** [review]
/approve

File: test/drivers/test_suites/syscall_exit_suite/accept_x.cpp [hunk 2]
```diff
@@ -210,7 +196,7 @@ TEST(SyscallExit, acceptX_UNIX)
 	evt_test->assert_numeric_param(4, (uint32_t)0);
 
 	/* Parameter 5: queuemax (type: PT_UINT32) */
-	evt_test->assert_numeric_param(5, (uint32_t)unix_max_queue_len);
+	evt_test->assert_numeric_param(5, (uint32_t)QUEUE_LENGTH);
```
Comment (hbrueckner) [inline test/drivers/test_suites/syscall_exit_suite/accept_x.cpp:199]: So looks like the "odd" behavior has gone now? At least on my system, changing to `QUEUE_LENGTH` now seems to work.

**Review by hbrueckner — APPROVED** [review]
Thanks @Andreagit97 
/approve

File: test/drivers/test_suites/syscall_exit_suite/accept_x.cpp (hunk 2 above)
Comment (Andreagit97) [reply test/drivers/test_suites/syscall_exit_suite/accept_x.cpp:199]: ahahahaha unfortunately i left the comment only in the big PR, but there was a reason behind it https://github.com/falcosecurity/libs/pull/783#discussion_r1044598112

**Pull request merged**